src/test/java/com/wikia/webdriver/common/templates/NewTestTemplate.java

```diff
@@ -19,11 +21,9 @@ public void beforeClass() {
 
   @BeforeMethod(alwaysRun = true)
   public void start(Method method, Object[] data) {
-    runProxyServerIfNeeded(method);
-    if (method.getAnnotation(UserAgent.class) != null) {
-      setBrowserUserAgent(
-          method.getAnnotation(UserAgent.class).userAgent()
-      );
+    String countryCode = getCountryCode(method, data);
+    if (countryCode != null || method.getAnnotation(NetworkTrafficDump.class) != null) {
+      runProxyServer(countryCode);
```

Comment:
my suggestion is to make a method boolean isProxyNeeded() or something like that, insted of checkin nulls and so on.